dot/parachain/dispute/backend_test.go

```diff
@@ -0,0 +1,153 @@
+package dispute
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/parachain/dispute/types"
+
+	parachainTypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/dgraph-io/badger/v4"
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/btree"
+)
+
+func TestOverlayBackend_EarliestSession(t *testing.T) {
+	t.Parallel()
+
+	// with
```

Comment:
What is this with, when, then?

Reply:
Just something I picked up from parity devs. `with` the give data, `when` we perform operations, `then` assert the responses!

Reply:
hmmm interesting I have not seen this before! I'm somewhat skeptical of having to many different ways of formatting tests, but unless others disagree seems okay to me

Reply:
Yeah, it's mostly used in ink!. This format can't be used for all the tests. It just made sense here.

Are we following some sort of standards for tests?

Reply:
Well cool thank you for informing me! 

I would say we do not at this moment, as we do different things in different parts of the code, so I think if this makes sense here I am good with it